peps/pep-0762.rst

```diff
@@ -0,0 +1,276 @@
+Author: Pablo Galindo Salgado <pablogsal@python.org>, Łukasz Langa <lukasz@python.org>, Lysandros Nikolaou <lisandrosnik@gmail.com>, Emily Morehouse-Valcarcel <emily@python.org>
+Sponsor: Pablo Galindo Salgado 
+Status: Final
+Type: Informational
+Created: 11-Oct-2024
+
+Abstract
+========
+
+One of Python’s core strengths is its interactive mode, also known as the
+Read-Eval-Print Loop (REPL), or the Python console, or the Python shell. This
+PEP describes the new implementation of this functionality written in Python.
+The new REPL released in Python 3.13 aims to provide modern features expected by
+today's users, such as multiline editing, syntax highlighting, custom commands,
+and an overall improved interactive experience.
+
+Motivation
+==========
+
+Up to Python 3.12, the interactive shell of CPython was written in C as a
+special mode of the parser. It was therefore difficult to maintain and extend.
+It relied on the existence of GNU readline (or an equivalent) for basic
+functionality like cursor movement and history tracking. Python compiled without
+this library provided an interactive mode of very limited capability. On the
+other hand, Python compiled with readline outsourced decisions and configuration
+around user input in ways that made extending it difficult.
+
+This complexity has deterred contributions and has made it challenging to
+implement new features. As a result, the CPython interactive shell has seen
+minimal changes, falling behind user expectations for modern equivalents.
+
+Many features that users have come to expect from modern REPLs were absent in
+the previous version. Some examples of these features include multi-line editing
+and history, custom commands, syntax highlighting, or ergonomic handling of copy
+and paste. The lack of these features greatly impacts the user experience of
+many groups of users of CPython, in particular in environments where users don’t
+control dependencies and cannot install their own packages. This is especially
+common for users learning the language and educators.
+
+Addressing such issues with the C implementation would require complex
+workarounds, such as AST matching of commands, which would add prohibitive
+complexity to the codebase.
+
+With the new REPL written in Python, we are addressing these limitations while
+also bringing CPython's interactive experience more in line with modern
+expectations and capabilities.
+
+Rationale
+=========
+
+Implementing the new REPL in Python, rather than C, has significantly lowered
+the barrier to entry for contributors. This change has made it easier to test,
+validate, and modify the REPL, leading to increased community involvement and
+faster feature development. The improved accessibility of the codebase is
+expected to result in a more rapidly evolving and user-responsive REPL.
+
+Instead of writing a Python REPL from scratch, we decided to base the
+implementation of the new REPL on PyREPL. This decision was driven by several
```

Comment:
Add a link to the PyREPL source code?

Reply:
Well now is challenging because PyPy adopted our version of PyREPL :sweat_smile: 

I can link to https://github.com/pypy/pyrepl but that's not the one we used

Reply:
I mostly feel it's important to link to something, because the PEP currently just says "PyREPL" with no background on what that is. I for one had never heard of the project before you added it to CPython, and I'm sure I'm not the only one :)

Reply:
Perhaps link to the commit that you worked from in the pyrepl repo.